app/models/evaluation.rb

```diff
@@ -231,13 +246,18 @@ def self.to_csv(json = nil)
     json_params = json.nil? ? nil : JSON.parse(json)
     filter_params = json_params["_json"].nil? ? nil : json_params["_json"]
 
-    if filter_params.nil?
-      # this will never be reached currently, need to fix.
-      evaluations = Evaluation.order(id: :asc)
-    else
-      where_params = parse_filters(filter_params)
-      evaluations = run_csv_query(where_params)
+    where_statement = []
+    where_params = parse_filters(filter_params)
+    where_params.map do |k, v|
+      where_statement << where_fields(k,v) unless v.nil?
     end
-    generate_csv(evaluations)
+
+    where_statement = where_statement.length > 1 ? where_statement.join(' AND ') : where_statement.first
```

Comment:
You should be able to do the join even with just one element in the array, or even the empty array.
The result of that will be the only value of the array or the empty string if the array is empty, so I think you can remove the conditional here

Reply:
Thanks @Levia I tried this, but now it breaks when there are no filters selected. I can rework the code however to work anyway in that case.

Reply:
`where_statement = where_statement.join(' AND ')` didnt' work you mean?

Reply:
you might need to do `where_statement.present?` instead of `.nil?` in the first line of the `generate_csv` method

Reply:
@Levia I have reworked that slightly now which is more in line with what you were suggesting so please take a look when you get a chance to!